version: '3.3'

services:
  redis_db:
    image: redis:alpine
    expose:
      - 6379
    volumes:
      - redis-data:/data
  db:
    image: postgres:15
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
    volumes:
      - ./postgres:/var/lib/postgresql/
      - ./postgres_data:/var/lib/postgresql/data/
    env_file:
      - ./.env.prod.db

  telegram_bot:
    build: ./telegram_bot
    restart: always
    volumes:
      - static_volume:/home/app/django_admin/staticfiles
      - media_volume:/home/app/django_admin/mediafiles
      - sqlite_database:/home/app/telegram_bot/database
      - ./files_configure:/home/app/telegram_bot/files_configure
      - ./loggs:/home/app/telegram_bot/loggs
    env_file:
      - ./.env.prod.telegram
    depends_on:
      - django_admin
      - db

  django_admin:
    build: ./django_admin
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - static_volume:/home/app/django_admin/staticfiles
      - media_volume:/home/app/django_admin/mediafiles
      - ./admin_logs:/home/app/django_admin/loggs
    expose:
      - 8000
    env_file:
      - ./.env.prod.django
    depends_on:
      - db
  nginx:
    build: ./nginx
    volumes:
      - static_volume:/home/app/django_admin/staticfiles
      - media_volume:/home/app/django_admin/mediafiles
    ports:
      - 80:80
    depends_on:
      - django_admin

volumes:
  redis-data:
  sqlite_database:
  static_volume:
  media_volume:
